server/schedule/checker/rule_checker.go

```diff
@@ -216,6 +216,13 @@ func (c *RuleChecker) fixRulePeer(region *core.RegionInfo, fit *placement.Region
 			ruleCheckerReplaceOfflineCounter.Inc()
 			return c.replaceUnexpectRulePeer(region, rf, fit, peer, offlineStatus)
 		}
+
+		if c.isWitnessEnabled() && c.isPendingVoter(region, peer) {
+			if witness, ok := c.hasAvailableWitness(region, peer); ok {
+				ruleCheckerPromoteWitnessCounter.Inc()
+				return operator.CreateNonWitnessPeerOperator("promote-witness", c.cluster, region, witness)
```

Comment:
How about telling apart fixing pending voter and fixing down peer since the operator are both "promote-witness"?